python/sglang/srt/model_loader/remote_instance_weight_loader_utils.py

```diff
@@ -67,3 +75,109 @@ def trigger_transferring_weights_request(
     except Exception as e:
         logger.error(f"Failed to trigger send weights to remote instance request: {e}")
         raise
+
+
+def get_remote_instance_transfer_engine_info_per_rank(seed_url: str, rank: int):
+    try:
+        response = requests.get(
+            f"{seed_url}/get_remote_instance_transfer_engine_info",
+            params={
+                "rank": rank,
+            },
+        )
+
+        if response.status_code == 200:
+            data = response.json()
+
+            if "remote_instance_transfer_engine_info" in data:
+                return data["remote_instance_transfer_engine_info"]
+            else:
+                logger.error(
+                    "Failed to get `remote_instance_transfer_engine_info` in response."
+                )
+                return None, None
+        else:
+            logger.error(f"request.get failed: {response.status_code}")
+            return None, None
+    except Exception as e:
+        logger.error(f"Exception: {e}")
+        return None, None
+
+
+def register_memory_region(model, transfer_engine):
+    start_tic = time.time()
+
+    weight_mr_dict = {}
+    for name, weight in model.named_parameters():
+        ret = transfer_engine.register_memory(
+            weight.data_ptr(), weight.numel() * weight.element_size()
+        )
+        if ret != 0:
+            raise RuntimeError(
+                f"register memory failed for weight {name}, error: {ret}"
+            )
+        weight_mr_dict[name] = (
+            weight.data_ptr(),
+            weight.numel(),
+            weight.element_size(),
+        )
+
+    end_tic = time.time()
+    logger.debug(f"Register memory region time: {(end_tic - start_tic):.4f}s")
+    return weight_mr_dict
```

Comment:
If this method is unnecessary for memory registration, remove it or mark it as deprecated.